internal/driver/driver.go

```diff
@@ -80,12 +81,22 @@ func (linodeDriver *LinodeDriver) SetupLinodeDriver(
 	linodeDriver.name = name
 	linodeDriver.vendorVersion = vendorVersion
 
+	// Validate the volume label prefix, if it is set.
+	//
+	// First, we want to make sure it is the right length, then we will make
+	// sure it only contains the acceptable characters.
+	//
+	// When checking the length, we will convert it to a []rune first, to count
+	// the number of unicode characters.
+	if r := []rune(bsPrefix); len(r) > MaxVolumeLabelPrefixLength {
+		return fmt.Errorf("volume label prefix is too long: length=%d max=%d", len(r), MaxVolumeLabelPrefixLength)
+	}
 	matched, err := regexp.MatchString(`^[0-9A-Za-z_-]{0,`+strconv.Itoa(MaxVolumeLabelPrefixLength)+`}$`, bsPrefix)
 	if err != nil {
-		return err
+		return fmt.Errorf("invalid regexp pattern: %w", err)
 	}
 	if !matched {
-		return fmt.Errorf("bs-prefix must be up to 12 alphanumeric characters, including hyphen and underscore")
+		return errors.New("volume label prefix may only contain: A-Z, a-z, 0-9")
```

Comment:
I think we still want to note that hyphens and underscores are acceptable right?

Reply:
:man_facepalming: Yep, I'll get that fixed up.